airflow/api_fastapi/core_api/routes/ui/structure.py

```diff
@@ -36,6 +38,10 @@
 @structure_router.get(
     "/structure_data",
     responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+    dependencies=[
+        Depends(requires_access_dag("GET")),
```

Comment:
In the legacy it was `@auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)`.

I think we need both:
-  `@auth.has_access_dag("GET", DagAccessEntity.TASK_INSTANCE)`
- `Depends(requires_access_dag("GET", DagAccessEntity.DEPENDENCIES))`

Because this is to retrieve the DAG graph (and if requested also external dependencies)

Reply:
I guess what you mean is `depends(requires_access_dag("GET", DagAccessEntity.TASK_INSTANCE))`?

Looking into the logic, I'm unsure I understand why we need this permission. 🤔 but I guess I missed some of the context from the UI end, so I just added it as well. Thanks!

Reply:
Thanks! I think the idea behind that in the legacy code is that this will return the graph dependencies. Tasks, setup/teardown, mapped, etc... With task ids and more information related to the tasks. So this requires TaskInstance permission on the dag.

Then we added the ability to request `external_dependencies` which will add external dependencies such as assets/aliases/trigger etc... (for this we need the Dependencies permission)